pros/cli/upload.py

```diff
@@ -83,6 +83,16 @@ def upload(path: Optional[str], project: Optional[c.Project], port: str, **kwarg
         raise dont_send(click.UsageError('No port provided or located. Make sure to specify --target if needed.'))
 
     if kwargs['target'] == 'v5':
+        slot = kwargs.get('slot', None)
+        if (slot == None):
+            try:
+                project_file = open(str(project.location)+"/project.pros", 'r')
+                import json
+                project_file_json = json.loads(project_file.read())
+                kwargs['slot'] = project_file_json['py/state']['upload_options']['slot']
```

Comment:
couple things here:

1. you should prefer to use the construct
```py
with f as open(filename):
    f.read()  # do something with this
``` 
2. you definitely don't need to read the JSON manually like this. all data that the CLI serializes is done in a python-compatible way using `jsonpickle`. also, we already have a project configuration abstraction-- see the first few lines of this method
3. now that i'm looking at this again, I think a better solution would be to modify lines 61-64 (instead of 86-95):
first, the original lines; problem highlighted:
```py
# apply project options as a template
options = dict(**project.upload_options)
options.update(kwargs)  # FIXME: this overwrites any project-specified upload slot with the default given by click
kwargs = options
```
here's a possible fix:
```py
# apply project options as a template
options = dict(**project.upload_options)

# check if slot is specified in project upload options but not in command line options
if 'slot' in options and kwargs.get('slot', None) is None:
    # default value for slot is None, so remove it
    # (click will place the default value of the flag in kwargs if it is unspecified, see #110 )
    kwargs.pop('slot')

# now we can just update options using kwargs-- if slot was specified at the command line it will override the project
# upload option, which is what we want
options.update(kwargs)
kwargs = options
```

Reply:
elliot and i had a spontaneous mind meld i guess. as elliot said, you may still need to set the default slot value to 1 later if it's still unspecified

Reply:
It needs an elif clause for when the option isn't present in the project file.
```
        if 'slot' in options and kwargs.get('slot', None) is None:
            # default value for slot is None, so remove it
            # (click will place the default value of the flag in kwargs if it is unspecified, see #110 )
            kwargs.pop('slot')
        elif (kwargs.get('slot', None) is None): 
            kwargs['slot'] = 1
```

Reply:
yeah that works. (also remember in python we don't parenthesize conditions like that)